### PR TITLE
175 create reusable component for scroll pagination for search different datas

### LINKED ISSUE #175: create reusable component for scroll pagination for search different datas
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Optional write device which you use**
...

**Additional context**

### PR DESCRIPTION
**Pull request by Vova-Romaniuk**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/users/maryanchuk04/projects/3)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: StudCity.Frontend/src/components/ScrollerPagination/index.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			if (indexPage === 1) {
+				setDatas(data.items)
+				setTotalItemsCount(data.pageViewModel.totalItemsCount)
+			}
+		}
+	}, [data.items])
+
+	const handleScroll = () => {
+		if (containerRef.current) {
+			if ((containerRef.current.clientHeight + containerRef.current.scrollTop > containerRef.current.scrollHeight - 1) && !loading) {
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/components/ScrollerPagination/index.jsx]: I would suggest before this if statement do some desctructuring of containerRef.current:

> const { clientHeight, scrollTop, scrollHeight } = contianerRef.current;

and then do if without duplicating contianerRef.current, this is purely for readability purposes

File: StudCity.Frontend/src/components/ScrollerPagination/index.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	data,
+	ItemComponent,
+	handleClick,
+	handleDispatch,
+	searchData,
+	selectedItems = []
+}) => {
+	const containerRef = useRef(null);
+	const [indexPage, setIndexPage] = useState(1);
+	const [datas, setDatas] = useState([]);
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/components/ScrollerPagination/index.jsx]: word 'data' is already means multiple items, you don't need an 's' at the end

**Review by Serhiy-dotcom — CHANGES_REQUESTED** [review]
pls fix small comments, overall seems good

File: StudCity.Frontend/src/components/ScrollerPagination/index.jsx (hunk 2 above)
Comment (maryanchuk04) [reply StudCity.Frontend/src/components/ScrollerPagination/index.jsx]: @Serhiy-dotcom @Vova-Romaniuk I think items will be better

File: StudCity.Frontend/src/components/SearchUsersForCreatedChats/index.jsx [hunk 3]
```diff
@@ -105,35 +53,23 @@ export default function UserGroupSearch({ setSelectedUsers, selectedUsers }) {
...
-					</Scroller>
+					<ScrollerPagination
+						loading={loading}
+						data={data}
+						handleDispatch={handleDispatch}
+						handleClick={handleClick}
+						searchData={searchData}
+						ItemComponent={User}
+						selectedItems={selectedUsers}
+					>
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/components/SearchUsersForCreatedChats/index.jsx]: use autoclosing component

**Review by maryanchuk04 — APPROVED** [review]

**Review by Serhiy-dotcom — APPROVED** [review]
Well done!

**Pull request merged**